### PR TITLE
fix(instagram): Updated to use new API

### PR DESCRIPTION
**Pull request by zachmccormick**
This fixes the Instagram provider to work with the new Instagram API as
per the docs at https://developers.facebook.com/docs/instagram-basic-display-api

# Submitting Pull Requests

## General

 - [x] Make sure you use [semantic commit messages](https://seesparkbox.com/foundry/semantic_commit_messages).
       Examples: `"fix(google): Fixed foobar bug"`, `"feat(accounts): Added foobar feature"`.
 - [x] All Python code must be 100% pep8 and isort clean.
 - [x] JavaScript code should adhere to [StandardJS](https://standardjs.com).
 - [x] If your changes are significant, please update `ChangeLog.rst`.
 - [x] Feel free to add yourself to `AUTHORS`.
 
 ## Provider Specifics
 
 In case you add a new provider:
 
- [ ] Make sure unit tests are available.
- [ ] Add an entry of your provider in `test_settings.py::INSTALLED_APPS` and `docs/installation.rst::INSTALLED_APPS`.
- [ ] Add documentation to `docs/providers.rst`.
- [ ] Add an entry to the list of supported providers over at `docs/overview.rst`.

### CODE REVIEW

File: allauth/socialaccount/providers/instagram/provider.py [hunk 1]
```diff
@@ -23,17 +20,16 @@ class InstagramProvider(OAuth2Provider):
     account_class = InstagramAccount
 
     def extract_extra_data(self, data):
-        return data.get('data', {})
```
Comment (zachmccormick) [inline allauth/socialaccount/providers/instagram/provider.py:26]: the shape no longer looks like

```json
{
  "data": {
    "id": "1234",
    "username": "foobar"
  }
}
```

File: allauth/socialaccount/providers/instagram/views.py [hunk 2]
```diff
@@ -13,11 +13,12 @@ class InstagramOAuth2Adapter(OAuth2Adapter):
     provider_id = InstagramProvider.id
     access_token_url = 'https://api.instagram.com/oauth/access_token'
     authorize_url = 'https://api.instagram.com/oauth/authorize'
-    profile_url = 'https://api.instagram.com/v1/users/self'
+    profile_url = 'https://graph.instagram.com/me'
```
Comment (zachmccormick) [inline allauth/socialaccount/providers/instagram/views.py:16]: https://developers.facebook.com/docs/instagram-basic-display-api/reference/me

File: allauth/socialaccount/providers/instagram/views.py [hunk 3]
```diff
@@ -13,11 +13,12 @@ class InstagramOAuth2Adapter(OAuth2Adapter):
...
 
     def complete_login(self, request, app, token, **kwargs):
         resp = requests.get(self.profile_url,
-                            params={'access_token': token.token})
+                            params={'access_token': token.token,
+                                    'fields': ['id', 'username']})
```
Comment (zachmccormick) [inline allauth/socialaccount/providers/instagram/views.py:21]: if you don't provide fields, you won't get the username back

**Comment by zachmccormick** [comment]
The only addition here is that we might want to change the default scope. I don't think it's `basic` anymore

**Comment by dorianfm** [comment]
@zachmccormick agree, the scope should be `user_profile` now, from what I can tell, instagram returns an error if `basic` scope is set.

**Comment by marcosalcazar** [comment]
+1 at this pull request, plus the required change on the scope as both @zachmccormick and @dorianfm have mentioned already. Thanks for the effort Zach, I'm using your branch for now.

**Comment by zachmccormick** [comment]
Ill try to update the scope and get this in a mergeable state this week unless someone else wants to make their own branch and get it mergeable

**Comment by pennersr** [comment]
> The only addition here is that we might want to change the default scope. I don't think it's `basic` anymore

https://www.instagram.com/developer/authorization/

Here they still mention `basic` as the only scope:

image

**Comment by dorianfm** [comment]
That’s the deprecated developer docs, you need to check out the ones on facebook.com for the current API

—

**Comment by dorianfm** [comment]
see info on migration of API here https://www.instagram.com/developer/

—

**Comment by mightym** [comment]
I just stumbled over this note while implementing the Instagram "Basic Display" in my django app:

> Note that Basic Display is not an authentication tool. Data returned by the API cannot be used to authenticate your app users or log them into your app. If your app uses API data to authenticate users, it will be rejected during App Review. If you need an authentication solution, use [Facebook Login](https://developers.facebook.com/docs/facebook-login/) instead.

So actually (as far as I understand this) Instagram isn't a way to authenticate a user anymore.

**Pull request merged**